File: python/datetime_parsing_csv/test_main.py
from pathlib import Path

import pytest
from main import read_and_modify, write_csv


@pytest.fixture
def persons():
    path_to_file = Path("./data/persons.csv")
    data = read_and_modify(path_to_file)
    return data


def test_read_and_modify(persons):
    assert len(persons) == 6
    assert persons[0].get("id") is not None


def test_write_csv(persons):
    output_path = Path("./data/test_persons.csv")
    write_csv(persons, output_path)
    assert output_path.exists()
    # Delete the file
    Path("./data/test_persons.csv").unlink()

File: rust/datetime_parsing_csv/src/test_main.rs
#![cfg(test)]

use std::path::Path;

#[test]
fn test_read_csv() {
    let input_path = Path::new("./data/persons.csv");
    let data = super::read_csv(input_path).expect("Unable to read/open CSV");
    assert_eq!(data.len(), 6);
}

#[test]
fn test_construct_person_obj() {
    let input_path = Path::new("./data/persons.csv");
    let data = super::read_csv(input_path).expect("Unable to read/open CSV");
    let persons_modified = super::construct_person_obj(data);
    assert_eq!(persons_modified.len(), 6);
    assert!(persons_modified[0].id > 0);
}

#[test]
fn test_write_csv() {
    let input_path = Path::new("./data/persons.csv");
    let persons = super::read_csv(input_path).expect("Unable to read/open CSV");
    let persons_modified = super::construct_person_obj(persons);
    let output_path = Path::new("./data/test_persons.csv");
    super::write_csv(persons_modified, output_path);
    assert!(output_path.exists());
    // Delete the file
    std::fs::remove_file(output_path).expect("Unable to delete output file");
}
